src/networkmanager/cni.cpp

```diff
@@ -182,8 +182,10 @@ Error CNI::SetConfDir(const String& configDir)
     mConfigDir = std::filesystem::path(configDir.CStr()) / "results";
 
     try {
-        if (!std::filesystem::create_directories(mConfigDir)) {
-            return Error(ErrorEnum::eFailed, "can't create directory");
+        if (!std::filesystem::exists(mConfigDir)) {
```

Comment:
You may not check if dir exists implicitly. create_directories does it for you.  Just do not check return value. If error occurs, create_directories will rise an exception.

Reply:
Done